linera-chain/src/inbox.rs

```diff
@@ -154,7 +154,9 @@ where
     }
 
     /// Consumes an event from the inbox.
-    pub(crate) async fn remove_event(&mut self, event: &Event) -> Result<(), InboxError> {
+    ///
+    /// Returns `true` if the event was already known, i.e. it was present in `added_events`.
+    pub(crate) async fn remove_event(&mut self, event: &Event) -> Result<bool, InboxError> {
```

Comment:
Optional: After looking at the tests, I'm wondering if it would make more sense to return a new type instead, something like `enum EventStatus { Queued, Processed }` (I might have gotten the variant names wrong though) :thinking:

Reply:
Would that be the same enum for `add_event` and `remove_event`? :thinking: 
It might be a bit ambiguous to return `Queued` from `remove_event` if the event was _removed_ from the queue.

Reply:
My initial thought was to use the same. But yeah, it might be hard to find the right name `{ Duplicate, FirstTime }` or something. (Not a blocker)